### PR TITLE
NAS-102425 / 11.3 / Expose netdata ips in netdata plugin

### PR DESCRIPTION
**Pull request by sonicaj**
This commit exposes netdata ips which user has configured and via which netdata can be accessed.

### CODE REVIEW

File: src/middlewared/middlewared/plugins/netdata.py [hunk 1]
```diff
@@ -37,6 +37,27 @@ def __init__(self, *args, **kwargs):
...
+    @accepts()
+    async def ips(self):
+        """
+        Returns a list of user configured addresses with which netdata can be accessed.
+        """
+        ips = []
+        for ip in (await self.config())['bind']:
+            if ip == '0.0.0.0':
+                ips.extend([
+                    f'http://{d["address"]}/netdata/'
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/netdata.py]: Why not https?

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/netdata.py]: `http+https` is enabled by default in the system. If user wants to enforce `https`, he/she can just set the relevant settings in system.general and all requests would be proxied to `https`.

File: src/middlewared/middlewared/plugins/netdata.py [hunk 2]
```diff
@@ -37,6 +37,27 @@ def __init__(self, *args, **kwargs):
             data['alarms'][alarm].pop('write_path', None)
         return data
 
+    @accepts()
+    async def ips(self):
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/netdata.py]: They are urls not IPs.

Also, what is the use case?

File: src/middlewared/middlewared/plugins/netdata.py (hunk 1 above)
Comment (themylogin) [reply src/middlewared/middlewared/plugins/netdata.py]: I mean we always have https enabled and we are trying to enforce to use it by default, why http is the default value here?

File: src/middlewared/middlewared/plugins/netdata.py (hunk 2 above)
Comment (sonicaj) [reply src/middlewared/middlewared/plugins/netdata.py]: UI requested to know with configurable binding ip's for netdata, where should they redirect the user to launch netdata.
And yes, i should change that to addresses. I wanted to wait till Monday tod discuss this with you as this has a potential problem with `nginx.conf` as there we direct any request with `/netdata` to localhost which in turn means that whichever addresses nginx is using, netdata will be available on those regardless of the fact if user only wants it on one ip.

File: src/middlewared/middlewared/plugins/netdata.py (hunk 1 above)
Comment (sonicaj) [reply src/middlewared/middlewared/plugins/netdata.py]: Copying `system.general` behavior where the default is to use `http` unless forced by the user. Please advise if you would like me to change this to `https`.

File: src/middlewared/middlewared/plugins/netdata.py (hunk 2 above)
Comment (william-gr) [reply src/middlewared/middlewared/plugins/netdata.py]: Sounds like thats not the solution.

They have the IP already from the browse, all they need is a path.

If they dont want to hardcode /netdata/ we can provide a method: `netdata.url_path`

Comment (william-gr) [reply src/middlewared/middlewared/plugins/netdata.py]: Also please request them to create a ticket

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/netdata.py]: Makes sense @william-gr. I think we should expose `netdata.url_path` after making sure that UI is okay with that else we can close this. Also i'll make sure in the future to request them to create a ticket first.

**Pull request closed without merging**